src/request.c

```diff
@@ -61,6 +61,14 @@ static uint8_t build_request_uri(CURL *curl, const char *base_domain,
   const char *http_protocol = "http";
   const char *https_protocol = "https";
   const char *protocol;
+  char port_str[32]; // ":portnumber" or empty string
```

Comment:
There's the `CURLOPT_PORT` option for `curl_easy_setopt` that would allow the port to be defined as a native integer type. Using this would seem like a better choice as it removes the need for type conversions.

Reply:
Oh, I wasn't aware of CURLOPT_PORT.  Yes, it seems like a better choice.  Should I go ahead and change the patch to use this?

Reply:
I'd say that would be a good idea as it would make the commit much smaller.

Reply:
Oh, I've ended up pushing the previous variant. Opened #92 for this.